### PR TITLE
ASGI cookbook recipe

### PR DESCRIPTION
**Pull request by erm**
Refers to https://github.com/Pylons/pyramid/issues/2603.

This provides an example ASGI WebSocket app using Pyramid. It also contains example deployment configurations for nginx and supervisor.

### CODE REVIEW

File: docs/deployment/asgi.rst [hunk 1]
```diff
@@ -0,0 +1,191 @@
+ASGI (Asynchronous Server Gateway Interface)
+++++++++++++++++++++++++++++++++++++++++++++
+
+
+This chapter contains information about using ASGI with
+Pyramid. You can read more about the specification here: https://github.com/django/asgiref/blob/master/specs/asgi.rst. 
+
+The example app below uses the WSGI to ASGI wrapper from the `asgiref` library to transform normal WSGI requests into ASGI responses - this allows the application to be run with an ASGI server, `uvicorn` or `daphne`.
```
Comment (stevepiercy) [inline docs/deployment/asgi.rst]: reStructuredText uses double-ticks for inline literals, single-ticks for italics/emphasis. If literals are your intent, then change them throughout, for example:

```rst
``asgiref``
```

However, when first mentioned in a narrative, it is a good idea to include links to references. Suggest for the first instance of each turning these into links, with subsequent mentions as inline literals.

```rst
`asgiref <https://pypi.org/project/asgiref/>`_
```

This is two sentences. "responses. This"

Finally:

```
 server, including `uvicorn <http://www.uvicorn.org/>`_ or `daphne <https://pypi.org/project/daphne/>`_.
```

Note, choosing appropriate URLs for a project depends on context. Here the context is docs. Unfortunately there are no ReadTheDocs URLs for these projects, so I picked the one that appears to best represent the project, with PyPI as the default. If you have better URLs, please use them instead.

File: docs/deployment/asgi.rst [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+The example app below uses the WSGI to ASGI wrapper from the `asgiref` library to transform normal WSGI requests into ASGI responses - this allows the application to be run with an ASGI server, `uvicorn` or `daphne`. 
+
+The example contains a class that extends the wrapper to enable routing ASGI consumers.
+
+
+Simple WSGI -> ASGI WebSocket application
+-----------------------------------------
+
+
```
Comment (stevepiercy) [inline docs/deployment/asgi.rst]: Remove 1 extra blank line.

File: docs/deployment/asgi.rst [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+    $ uvicorn app:app
+
+or 
+
+.. code-block:: bash
+
+    $ daphne app:app
+
+
+There are several potential deployment options, one example is `nginx` and `supervisor`.
```
Comment (stevepiercy) [inline docs/deployment/asgi.rst]:
```
`nginx <https://nginx.org/>`_ and `supervisor <http://supervisord.org/>`_.
```

**Comment by erm** [comment]
Thanks for the quick feedback :) 

I'll have a think about how I should go about hosting the documentation and then update the PR based on your suggestions.

**Comment by stevepiercy** [comment]
Thank you!

**Pull request merged**